src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimCallbackHandler.java

```diff
@@ -159,31 +191,99 @@ private CallbackResponse setApplicantResponseDeadline(CallbackParams callbackPar
+                .primaryAddress(caseData.getRespondent2Copy().getPrimaryAddress())
+                .build();
 
-        updatedData.respondent1DQ(dq);
-        // resetting statement of truth to make sure it's empty the next time it appears in the UI.
-        updatedData.uiStatementOfTruth(StatementOfTruth.builder().build());
+            updatedData.respondent2(updatedRespondent2).respondent2Copy(null);
+        }
+
+        LocalDateTime responseDate = time.now();
+        AllocatedTrack allocatedTrack = caseData.getAllocatedTrack();
+
+        // same legal rep - will respond for both and set applicant 1 response deadline
+        if (respondent2HasSameLegalRep(caseData)) {
+            // if responses are marked as same, copy respondent 1 values into respondent 2
+            if (caseData.getRespondentResponseIsSame() != null && caseData.getRespondentResponseIsSame() == YES) {
+                updatedData.respondent2ClaimResponseType(caseData.getRespondent1ClaimResponseType());
+            }
+
+            updatedData
+                .businessProcess(BusinessProcess.ready(DEFENDANT_RESPONSE))
+                .respondent1ResponseDate(responseDate)
+                .respondent2ResponseDate(responseDate)
+                .applicant1ResponseDeadline(getApplicant1ResponseDeadline(responseDate, allocatedTrack));
+
+            // moving statement of truth value to correct field, this was not possible in mid event.
+            StatementOfTruth statementOfTruth = caseData.getUiStatementOfTruth();
+            Respondent1DQ dq = caseData.getRespondent1DQ().toBuilder()
+                .respondent1DQStatementOfTruth(statementOfTruth)
+                .build();
+
+            updatedData.respondent1DQ(dq);
+            // resetting statement of truth to make sure it's empty the next time it appears in the UI.
+            updatedData.uiStatementOfTruth(StatementOfTruth.builder().build());
+
+            // only represents 2nd respondent - need to wait for respondent 1 before setting applicant response deadline
+        } else if (solicitorRepresentsOnlyRespondent2(callbackParams)) {
+            updatedData.respondent2ResponseDate(responseDate);
+
+            if (caseData.getRespondent1ResponseDate() != null) {
+                updatedData
+                    .applicant1ResponseDeadline(getApplicant1ResponseDeadline(responseDate, allocatedTrack))
+                    .businessProcess(BusinessProcess.ready(DEFENDANT_RESPONSE));
+            }
+            // represents 1st respondent - need to set deadline if only 1 respondent, or wait for 2nd respondent
+            // response before setting deadline
+        } else {
+            updatedData.respondent1ResponseDate(responseDate);
+
+            if ((caseData.getAddRespondent2() != null && caseData.getAddRespondent2() == NO)
+                || caseData.getRespondent2ResponseDate() != null
+                || (caseData.getAddApplicant2() != null && caseData.getAddApplicant2() == YES)) {
```

Comment:
So having a second applicant on the case alone is sufficient for this whole condition to evaluate to `TRUE` and therefore the case progresses by setting the deadline  - is that expected behaviour or an oversight?